Riot/Managers/Theme/ElementUIColorsResolved.swift

```diff
@@ -0,0 +1,91 @@
+    /// The colors from DesignKit, resolved for dark mode only.
+    static let elementDark = ElementUIColorsResolved(dynamicColors: element, userInterfaceStyle: .dark)
+}
+
+/// The dynamic colors from DesignKit, resolved to light or dark mode for use in the UIKit themes.
+///
+/// As Element doesn't (currently) update the app's `UIUserInterfaceStyle` when selecting
+/// a custom theme, the dynamic colors provided by DesignKit need resolving for each theme to
+/// prevent them from respecting the interface style and rendering in the wrong style.
+@objcMembers public class ElementUIColorsResolved: NSObject {
+    // MARK: Compound
+    public let accent: UIColor
+    public let alert: UIColor
+    public let primaryContent: UIColor
+    public let secondaryContent: UIColor
+    public let tertiaryContent: UIColor
+    public let quaternaryContent: UIColor
+    public let quinaryContent: UIColor
+    public let system: UIColor
+    public let background: UIColor
+    
+    public let namesAndAvatars: [UIColor]
+    
+    // MARK: Legacy
+    public let quarterlyContent: UIColor
+    public let navigation: UIColor
+    public let tile: UIColor
+    public let separator: UIColor
+    
+    // MARK: Setup
+    public init(dynamicColors: ElementUIColors, userInterfaceStyle: UIUserInterfaceStyle) {
+        let traitCollection = UITraitCollection(userInterfaceStyle: userInterfaceStyle)
+        
+        self.accent = dynamicColors.accent.resolvedColor(with: traitCollection)
+        self.alert = dynamicColors.alert.resolvedColor(with: traitCollection)
+        self.primaryContent = dynamicColors.primaryContent.resolvedColor(with: traitCollection)
+        self.secondaryContent = dynamicColors.secondaryContent.resolvedColor(with: traitCollection)
+        self.tertiaryContent = dynamicColors.tertiaryContent.resolvedColor(with: traitCollection)
+        self.quaternaryContent = dynamicColors.quaternaryContent.resolvedColor(with: traitCollection)
+        self.quinaryContent = dynamicColors.quinaryContent.resolvedColor(with: traitCollection)
+        self.system = dynamicColors.system.resolvedColor(with: traitCollection)
+        self.background = dynamicColors.background.resolvedColor(with: traitCollection)
+        
+        self.namesAndAvatars = [
+            dynamicColors.globalAzure.resolvedColor(with: traitCollection),
+            dynamicColors.globalGrape.resolvedColor(with: traitCollection),
+            dynamicColors.globalVerde.resolvedColor(with: traitCollection),
+            dynamicColors.globalPolly.resolvedColor(with: traitCollection),
+            dynamicColors.globalMelon.resolvedColor(with: traitCollection),
+            dynamicColors.globalAqua.resolvedColor(with: traitCollection),
+            dynamicColors.globalPrune.resolvedColor(with: traitCollection),
+            dynamicColors.globalKiwi.resolvedColor(with: traitCollection)
+        ]
```

Comment:
Ideally this array will be generated in the DesignKit package, but we can create them locally for now.